### PR TITLE
tailscale: add support for Darwin

### PR DESCRIPTION
**Pull request by pborzenkov**
###### Motivation for this change
Open source tailscale client is supported on Darwin and works just fine:
https://github.com/tailscale/tailscale/wiki/Tailscaled-on-macOS

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by danderson — APPROVED** [review]
Looks reasonable, thanks!

File: pkgs/servers/tailscale/default.nix [hunk 1]
```diff
@@ -36,7 +36,7 @@ buildGoModule rec {
   meta = with lib; {
     homepage = "https://tailscale.com";
     description = "The node agent for Tailscale, a mesh VPN built on WireGuard";
-    platforms = platforms.linux;
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/servers/tailscale/default.nix:39]:
```suggestion
```
default for buildGoModule

File: pkgs/servers/tailscale/default.nix [hunk 2]
```diff
@@ -36,7 +36,6 @@ buildGoModule rec {
   meta = with lib; {
     homepage = "https://tailscale.com";
     description = "The node agent for Tailscale, a mesh VPN built on WireGuard";
-    platforms = platforms.linux;
```
Comment (kalbasit) [inline pkgs/servers/tailscale/default.nix:39]: Why was this removed? Shouldn't this be `platforms.linux ++ platforms.darwin`?

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 140058` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tailscale</li>
  </ul>
</details>

File: pkgs/servers/tailscale/default.nix (hunk 2 above)
Comment (smancill) [reply pkgs/servers/tailscale/default.nix:39]: See: https://github.com/NixOS/nixpkgs/pull/140058#discussion_r719596196

Comment (kalbasit) [reply pkgs/servers/tailscale/default.nix:39]: Oh right :facepalm:
